test/components/views/rooms/RoomHeader-test.tsx

```diff
@@ -307,3 +333,395 @@ function findImg(wrapper: ReactWrapper, selector: string): ReactWrapper {
+                <RoomHeader
+                    room={room}
+                    inRoom={true}
+                    onSearchClick={() => { }}
+                    onInviteClick={null}
+                    onForgetClick={() => { }}
+                    onAppsClick={() => { }}
+                    e2eStatus={E2EStatus.Normal}
+                    appsShown={true}
+                    searchInfo={{
+                        searchTerm: "",
+                        searchScope: SearchScope.Room,
+                        searchCount: 0,
+                    }}
+                    {...props}
+                />
+            </RoomContext.Provider>,
+        );
+    };
+
+    it("hides call buttons in video rooms", () => {
+        mockRoomType(RoomType.UnstableCall);
+        mockEnabledSettings(["showCallButtonsInComposer", "feature_video_rooms", "feature_element_call_video_rooms"]);
+
+        renderHeader();
+        expect(screen.queryByRole("button", { name: /call/i })).toBeNull();
+    });
+
+    it("hides call buttons if showCallButtonsInComposer is disabled", () => {
+        mockEnabledSettings([]);
+
+        renderHeader();
+        expect(screen.queryByRole("button", { name: /call/i })).toBeNull();
+    });
+
+    it(
+        "hides the voice call button and disables the video call button if configured to use Element Call exclusively "
+        + "and there's an ongoing call",
+        async () => {
+            mockEnabledSettings(["showCallButtonsInComposer", "feature_group_calls"]);
+            SdkConfig.put({ element_call: { url: "https://call.element.io", use_exclusively: true } });
+            await ElementCall.create(room);
+
+            renderHeader();
+            expect(screen.queryByRole("button", { name: "Voice call" })).toBeNull();
+            expect(screen.getByRole("button", { name: "Video call" })).toHaveAttribute("aria-disabled", "true");
+        },
+    );
+
+    it(
+        "hides the voice call button and starts Element calls when the video call button is pressed if configured to "
+        + "use Element Call exclusively",
+        async () => {
+            mockEnabledSettings(["showCallButtonsInComposer", "feature_group_calls"]);
+            SdkConfig.put({ element_call: { url: "https://call.element.io", use_exclusively: true } });
+
+            renderHeader();
+            expect(screen.queryByRole("button", { name: "Voice call" })).toBeNull();
+
+            const dispatcherSpy = jest.fn();
+            const dispatcherRef = defaultDispatcher.register(dispatcherSpy);
+            fireEvent.click(screen.getByRole("button", { name: "Video call" }));
+            await waitFor(() => expect(dispatcherSpy).toHaveBeenCalledWith({
+                action: Action.ViewRoom,
+                room_id: room.roomId,
+                view_call: true,
+            }));
+            defaultDispatcher.unregister(dispatcherRef);
+        },
+    );
+
+    it(
```

Comment:
Why should the video button be disabled?

Reply:
Because in this scenario the user lacks permission to use it, and Gaëlle has told me that we should expose context like that on the button directly, rather than letting the user try to use the button but ending up with an error modal